front/pages/api/w/[wId]/subscriptions/stripe_subscription_valid.ts

```diff
@@ -0,0 +1,188 @@
+import type { WithAPIErrorReponse } from "@dust-tt/types";
+import type { NextApiRequest, NextApiResponse } from "next";
+import type Stripe from "stripe";
+
+import { Authenticator, getSession } from "@app/lib/auth";
+import { getStripeSubscription } from "@app/lib/plans/stripe";
+import { apiError, withLogging } from "@app/logger/withlogging";
+
+// TODO(2024-04-05,pr): immediately after flav's merge, use the global constant
+const REPORT_USAGE_VALUES = ["MAU_1", "MAU_5", "MAU_10", "PER_SEAT"];
+
+export type GetSubscriptionResponseBody =
+  | { valid: true }
+  | { valid: false; reason: string };
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorReponse<GetSubscriptionResponseBody>>
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const stripeSubscriptionId = req.query.stripeSubscriptionId as string;
+  if (!stripeSubscriptionId || typeof stripeSubscriptionId !== "string") {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid request, missing stripeSubscriptionId.",
+      },
+    });
+  }
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace was not found.",
+      },
+    });
+  }
+
+  // Callers will be either superUsers (from poke) or admins of the workspace (from webhook calls)
+  if (!auth.isAdmin()) {
+    return apiError(req, res, {
+      status_code: 403,
+      api_error: {
+        type: "workspace_auth_error",
+        message:
+          "Only users that are `admins` for the current workspace can access this endpoint.",
+      },
+    });
+  }
+
+  // for admins of the workspace (webhook call), we must already have the
+  // subscription in our DB (matching the workspace)
+  const subscription = auth.subscription();
```

Comment:
Do we plan to expose this endpoint to our users?

Reply:
Probably not yet; might happen in the future

But we plan to call it from webhook immediately, in which we are not auth'd as superuser (and shouldn't be IIRC)

Reply:
If we only need pole and webhooks I’d be in favour of factorising as much as possible in lib/api and have a poke only route + call from the webhook directly ?